clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -324,11 +332,35 @@ public synchronized boolean add(final MetricName metricName, final MeasurableSta
             );
             registry.registerMetric(metric);
             metrics.put(metric.metricName(), metric);
-            stats.add(new StatAndConfig(Objects.requireNonNull(stat), statConfig));
+            stats.add(new StatAndConfig(Objects.requireNonNull(stat), statConfig, metricName));
             return true;
         }
     }
 
+    /**
+     * Update config of a measurable stat and metric registered with this sensor that corresponds to a given metric name
+     * @param metricName The name of the metric to update
+     * @param config     New configuration for this metric
+     * @return true if config was updated, false if sensor expired or metric is not registered with the sensor
+     */
+    public synchronized boolean update(final MetricName metricName, final MetricConfig config) {
```

Comment:
nit: The method signature for StatAndConfig has MetricConfig parameter first and MetricName second. This one has the reverse order.